### PR TITLE
Pull request for DATACASS-201

### PR DESCRIPTION
**Pull request by sksumit1**
DATACASS-201

### CODE REVIEW

**Comment by sksumit1** [comment]
Issue created https://jira.spring.io/browse/DATACASS-201

File: spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/CassandraOperations.java [hunk 1]
```diff
@@ -280,7 +281,40 @@
...
+	<T> List<T> update(Update update, Class<T> type);
+	/**
+	 * Execute the Update Query and convert to the list of entities
+	 * 
+	 * @param Update must not be {@literal null}.
+	 * @param type must not be {@literal null}, mapped entity type.
+	 * @param options The {@link WriteOptions} to use.
+	 * @return
+	 */
+	<T> List<T> update(Update update, Class<T> type, WriteOptions options);
```
Comment (mp911de) [inline spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/CassandraOperations.java]: `WriteOptions` (ttl, consistency level, retry policy) are already part of the `Update` object so I'd propose to skip this method.

File: spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/CassandraOperations.java [hunk 2]
```diff
@@ -280,7 +281,40 @@
...
+	<T> List<T> update(List<Update> updateList, Class<T> type);
+	/**
+	 * Execute the Update Query list and convert to the list of entities
+	 * 
+	 * @param Update must not be {@literal null}.
+	 * @param type must not be {@literal null}, mapped entity type.
+	 * @param options The {@link WriteOptions} to use.
+	 * @return
+	 */
+	<T> List<T> update(List<Update> updateList, Class<T> type, WriteOptions options);
```
Comment (mp911de) [inline spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/CassandraOperations.java]: `WriteOptions` (ttl, consistency level, retry policy) are already part of the `Update` object so I'd propose to skip this method.

**Comment by mp911de** [comment]
@sksumit1 sorry for the long radio silence. Thanks for your PR, are you still interested to continue on this PR?

**Comment by sksumit1** [comment]
Yes, sure. Should i get rid of 
<T> List<T> update(List<Update> updateList, Class<T> type, WriteOptions options)

**Comment by mp911de** [comment]
Yes, please remove both methods specifying `WriteOptions` from your PR. `Update`/`List<Update>` and `WriteOptions` specify a common subset of options which leads to confusion what options to use.

**Comment by sksumit1** [comment]
ok...let me come back with the changes

**Comment by sksumit1** [comment]
Changes done, tests passing

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] Spring Data Cassandra .............................. SUCCESS [  0.350 s]
[INFO] Spring CQL ......................................... SUCCESS [  9.055 s]
[INFO] Spring Data Cassandra - Core ....................... SUCCESS [  3.156 s]
[INFO] Spring Data Cassandra - Distribution ............... SUCCESS [  0.008 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS

**Comment by mp911de** [comment]
Thanks again for coming back with changes. I took another look at the PR. I have two general issues:

1. `Update` commands do not return a result in the `ResultSet` so passing in a return type and returning a list of entities is pointless
2. Spring Data Cassandra is running behind the scenes a batch. This is fine for certain scenarios. In other cases, it creates a lot of work for the coordinator. I think this shouldn't happen as these methods hide the batching aspect. It's rather a surprise to the user that batching happens here.

I created two tickets based on your PR to provide batch operations:
* https://jira.spring.io/browse/DATACASS-288
* https://jira.spring.io/browse/DATACASS-289

I'm still new to the Cassandra project and had less an idea when picking up this PR. I have now a better overview of the project and a better idea of the concepts behind Spring Data Cassandra. 

I'd like to close this PR and continue efforts based on the newly created tickets. A dedicated batching API expresses the real intent based and your changes (execution of batch-updates) would fit into the batching API concept.

**Comment by mp911de** [comment]
Closing this one as we're following up the topic in other tickets.

**Pull request closed without merging**